### DISCUSSION TITLE
how can I generate input file for a certain aircraft?

### DISCUSSION THREAD

**Discussion by sarapoT** [ImperialCollegeLondon/sharpy#226] [Q&A]
Hi!
I wanna ask :is there an easier way to generate my own iput files for another hale uav,or I have to write my own generae-xxxx.py in a certain format?

**Comment by ACea15** [comment]
Hi, you have plenty of examples on how to generate cases in the tests folder and in the Python notebooks: sharpy/docs/source/content/example_notebooks/. The process is carried out via a python file that generates the input files in h5 format.  
We actually have a tool for model generation that is much simpler and only requires a python dictionary to generate the models; it is however in beta, but we could push it to the develop branch for you to play with it, please post the kind of problem that you want to model and we would assess the best approach for it.

**Reply by sarapoT** [reply]
Hi！Thank u so much for your reply！I`m working on aeroelasitc analysis of a twin-body Hale plane, including fluter ananlysis and gust response analysis.  It would be much easier to have a generator module !  Where can I find this branch?

**Comment by ACea15** [comment]
Ok so the branch is dev_rhea, the approach is based on components that are defined independently and automatically assembled together. 
have a look at these examples, it is the hale that becomes fully parameterised:
https://github.com/ACea15/Projects-SHARPy/blob/dev_rhea/aeroelasticPMOR_Optimization/parametric_aircraft/parametric_hale.py

as you can see the idea is that you define a dictionary of components,
g1c = dict()
g1c['fuselage'] = {'workflow':['create_structure','create_aero0'],  #creates structure but no aero
                  'geometry': {'length':10,
                               'num_node':11,
                               'direction':[1.,0.,0.],
                               'sweep':0.,
                               'dihedral':0.},
                  'fem': {'stiffness_db':stiffness,
                          'mass_db':mass_fuselage,
                          'frame_of_reference_delta':[0,1.,0.],
                          'lumped_mass': lumped_mass,
                          'lumped_mass_nodes':lumped_mass_nodes,
                          'lumped_mass_inertia':lumped_mass_inertia,
                          'lumped_mass_position':lumped_mass_position}
}

g1c['wing_r'] = {'workflow':['create_structure', 'create_aero'],
                  'geometry': {'length':20.,
                               'num_node':11,
                               'direction':[0.,1.,0.],
                               'sweep':20.*np.pi/180,
                               'dihedral':0.},
                  'fem': {'stiffness_db':stiffness,
                          'mass_db':mass_wing,
                          'frame_of_reference_delta':[-1, 0., 0.]},
                  'aero': {'chord':[1.,1.],
                           'elastic_axis':0.33,
                           'surface_m':16}
                 }
.
.
.

then tell how the components are put together and whether we want to have a factory of parametric models with the variables defining each component,

g1mm = {'model_name':'hale_test',
        'model_route':model_route,
        'iterate_type': 'Full_Factorial',
        'write_iterate_vars': True,
        'iterate_vars': {'fuselage\*geometry-length':np.linspace(7,15.,3),          #full factorial models varying these variables
                         'wing_r\*geometry-length':np.linspace(15,25.,3),
                         'winglet_r\*geometry-dihedral':np.pi/180*np.array([0,20,40])},
        'iterate_labels': {'label_type':'number',
                           'print_name_var':0},
        'assembly': {'include_aero':1,     #very simple since all definitions are local, the global assemble of fem and aero is taken care of.
                     'default_settings': 1, 
                     'fuselage':{'upstream_component':'',
                                 'node_in_upstream':0},
                     'wing_r':{'keep_aero_node':1,
                               'upstream_component':'fuselage',
                               'node_in_upstream':0},
                     'winglet_r':{'keep_aero_node':1,
                               'upstream_component':'wing_r',
                               'node_in_upstream':10},
...

another thing that was being developed was a generator of solutions (trim, flutter, gusts...) with minimal parameters in a sort of Nastran aeroelastic fashion. You can find examples in the link above. Honestly all this work has been in stand-by (and some things still need work) but  we might consider resuming development if there is a need for it.

**Reply by nishanttn** [reply]
Hi, is there a way to implement control surface in this above program to run static trim solver?

**Comment by sarapoT** [comment]
hi！ I met another question and i dont konow how to fix it:  I runned the example “cases/coupled/simple_HALE” case in a docker environment，everything went on well，but I can`t find the "timestep_info" or "case_data.structure"or "case_data.structure".   I checked another example file"generate_XHALE" and found that adding 'WriteVariablesTime' can make it output some  time-dependant variables, like velocity and acceleration. but some virables like wing -root-loads,i dont know how to generate it..    SO how can I make the program output "timestep_info"/ "case_data.structure" /"case_data.structure" directly?

**Comment by ACea15** [comment]
I believe PickleData postprocessor might do the job for you. If you do a grep in the repo you will find all the example cases were it is used.
